runtime/compiler/compile/J9Compilation.cpp

```diff
@@ -77,23 +76,31 @@ void *operator new(size_t size)
    if (firstCompileStarted)
    #endif
       {
+#if !defined(JITSERVER_SUPPORT)
       printf( "\n*** ERROR *** Invalid use of global operator new\n");
       TR_ASSERT(0,"Invalid use of global operator new");
+#endif
       }
 #endif
-   return malloc(size);
+   void* pMemory = malloc(size);
+   if (pMemory)
+      return pMemory;
+   else
+      throw std::bad_alloc();
    }
 
 /**
  * Since we are using arena allocation, heap deletions must be a no-op, and
- * can't be used by JIT code, so we inject an assertion here.
+ * can't be used by JIT code (JITServer is an exception as protobuf uses new and delete), 
+ * so we inject an assertion here.
  */
-void operator delete(void *)
+void operator delete(void * pMemory) throw()
    {
+#if !defined(JITSERVER_SUPPORT)
```

Comment:
For operator new we had a test on `firstCompileStarted` because "glibc allocates something at dl_init". If glibc could deallocate as well we need a similar change here.